internal/controllers/cluster/cluster_controller.go

```diff
@@ -87,6 +87,19 @@ type Reconciler struct {
 }
 
 func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	if r.Client == nil {
```

Comment:
Let's go with this instead
```
	if r.Client == nil || r.APIReader == nil || r.ClusterCache == nil || r.RemoteConnectionGracePeriod == time.Duration(0) {
		return errors.New("Client, APIReader and ClusterCache must not be nil and RemoteConnectionGracePeriod must not be 0")
	}
```

* I think it makes sense to use the exact upper case field names here, as that is what folks would have to set
* Let's try to keep this pretty compact to not add too much noise to our SetupWithManager funcs (in general these errors should only ever be seen by developers until they fixup their setup, so it's not super important to have a very precise error message that contains the exact field that is misconfigured)

Reply:
Updated as suggested.